sky/global_user_state.py

```diff
@@ -107,13 +119,35 @@ class StorageStatus(enum.Enum):
 def add_or_update_cluster(cluster_name: str,
                           cluster_handle: 'backends.Backend.ResourceHandle',
                           ready: bool,
-                          is_launch: bool = True):
+                          is_launch: bool = True,
+                          task=None):
     """Adds or updates cluster_name -> cluster_handle mapping."""
     # FIXME: launched_at will be changed when `sky launch -c` is called.
     handle = pickle.dumps(cluster_handle)
     cluster_launched_at = int(time.time()) if is_launch else None
     last_use = common_utils.get_pretty_entry_point() if is_launch else None
     status = ClusterStatus.UP if ready else ClusterStatus.INIT
+
+    cluster_hash = get_cluster_hash(cluster_name)
+
+    if not cluster_hash:
+        cluster_hash = str(uuid.uuid4())
+
+    launched_resources = get_launched_resources(cluster_handle)
+
+    requested_resources = None
+    if task:
+        assert len(task.resources) == 1, task.resources
+        requested_resources = list(task.resources)[0]
```

Comment:
This will update the `requested_resources` whenever the `sky launch` is called. We need to make sure it is only set at the first time the cluster is launched, i.e. only do it when `usage_intervals` is empty.